diff --git a/.cipd_impl.ps1 b/.cipd_impl.ps1
index eb837513..2d6bfe67 100644
--- a/.cipd_impl.ps1
+++ b/.cipd_impl.ps1
@@ -101,6 +101,11 @@ function Download-File {
 
   $wc = (New-Object System.Net.WebClient)
   $wc.Headers.Add("User-Agent", $UserAgent)
+  $wc = (New-Object System.Net.WebClient)
+  $proxy = (new-object System.Net.WebProxy)
+  $proxy.Address = "http://127.0.0.1:9798"
+  $wc.Headers.Add("User-Agent", $UserAgent)
+  $wc.Proxy = $proxy
   try {
     # Download failures were reported on Windows 8.1 without this line.
     [System.Net.ServicePointManager]::SecurityProtocol = `
@@ -116,7 +121,6 @@ function Download-File {
   }
 }
 
-
 # Retry a command with a delay between each.
 function Retry-Command {
   [CmdletBinding()]
